dts/arm/silabs/efr32bg22.dtsi

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2021 Sateesh Kotapati
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv8-m.dtsi>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/i2c/i2c.h>
+#include <dt-bindings/pinctrl/gecko-pinctrl.h>
+#include "gpio_gecko.h"
+
+/ {
+	chosen {
+		zephyr,flash-controller = &msc;
+	};
+
+	power-states {
+		standby: standby {
+			compatible = "zephyr,power-state";
+			power-state-name = "standby";
+			min-residency-us = <50000>;
+		};
+	};
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m33";
+			reg = <0>;
+			cpu-power-states = < &standby >;
+		};
+	};
+
+	sram0: memory@20000000 {
+		compatible = "mmio-sram";
+	};
+
+	soc {
+		msc: flash-controller@40030000 {
+			compatible = "silabs,gecko-flash-controller";
+			reg = <0x40030000 0xC69>;
+			interrupts = <49 0>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				write-block-size = <4>;
+				erase-block-size = <8192>;
+			};
+		};
+
+		usart0: usart@4005c000 { /* USART0 */
+			compatible = "silabs,gecko-usart";
+			reg = <0x4005C000 0x400>;
+			interrupts = <13 0>, <14 0>;
+			interrupt-names = "rx", "tx";
+			peripheral-id = <0>;
+			status = "disabled";
+		};
+
+		usart1: usart@40060000 { /* USART1 */
+			compatible = "silabs,gecko-usart";
+			reg = <0x40060000 0x400>;
+			interrupts = <15 0>, <16 0>;
+			interrupt-names = "rx", "tx";
+			peripheral-id = <1>;
+			status = "disabled";
+		};
+
+		leuart0: leuart@4004a000 { /* LEUART0 */
+			compatible = "silabs,gecko-leuart";
+			reg = <0x4004a000 0x400>;
+			interrupts = <22 0>;
+			peripheral-id = <0>;
```

Comment:
What is this "peripheral-id" used for?

Reply:
It's used for indexing if there are multiple peripherals of given type. For example in GPIO driver it is used for getting `gpio_index`. In LEUART it is used to distinguish between `cmuClock_LEUART0` and `cmuClock_LEUART1` clocks.

Reply:
Here is the usage of this property in [GPIO driver](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/gpio/gpio_gecko.c#L397) and in [LEUART driver usage#0](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/leuart_gecko.c#L330) + [LEUART driver usage#1](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/leuart_gecko.c#L383).
You can also see this in [UART driver](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_gecko.c#L538), [SPI driver](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/spi/spi_gecko.c#L332) and [Watchdog driver](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/watchdog/wdt_gecko.c#L284).

It seems that we always use this property alongside the instance ID but if we start using it instead of `peripheral_id` from DTS then e.g. UART driver won't work at all so in the case of `uart_gecko.c` the driver instance ID is not the same as `perpheral_id` from DTS nodes.

Reply:
As suspected, this is used because of how HAL works (it takes index 0 for gpioa, 1 for gpiob, etc.). Ref https://github.com/antmicro/hal_silabs/blob/bg22-minimal/gecko/emlib/inc/em_gpio.h#L1045. So, this is not hardware description, but DT HALization. You can create a match table in your driver between DT entries and HAL expectations (or don't use HALs :-) ).

cc: @galak

Reply:
Yes, so I marked `peripheral-id` property in GPIO and UART bindings as not required and modified drivers to not use this property when it's not defined for given node.